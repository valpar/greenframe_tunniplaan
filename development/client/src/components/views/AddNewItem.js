import Modal from "../UI/Modal/Modal";
import NewSubject from "./inputRows/NewSubject";
import { useEffect, useState } from "react";
import axios from "axios";
import NewLecturer from "./inputRows/NewLecturer";
import NewRoom from "./inputRows/NewRoom";
import NewCourse from "./inputRows/NewCourse";
import ConfirmModal from "../UI/ConfirmModal/ConfirmModal";
import TooltipLarge from "../UI/Tooltip/TooltipLarge";
import content from "../../assets/content/content.json";
import RequestModal from "../UI/RequestModal/RequestModal";

axios.defaults.baseURL = `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`;

const AddNewItem = (props) => {
  const {
    modalFor,
    onNewItem,
    scheduled,
    roomsData,
    courseData,
    teacherData,
    subjectsData,
    onClose,
    onDelete,
    editMode,
    editValues,
  } = props;
  const {
    deleteMessage,
    roomHasActiveTeachers,
    courseHasActiveTeachers,
    teacherHasActiveTeachers,
    subjectHasActiveTeachers,
    saveMessage,
  } = content.confirmModalMessages;

  const token = JSON.parse(sessionStorage.getItem('token'));

  const { mandatoryFields } = content.errorMessages;

  const [inputsState, setInputsState] = useState([{}]);
  const [inputsAreValid, setInputsAreValid] = useState([{ inputs: false }]);
  const [validSubmit, setValidSubmit] = useState(true);
  const [responseId, setResponseId] = useState();
  const [showDeleteConfirmModal, setShowDeleteConfirmModal] = useState(false);
  const [showUpdateConfirmModal, setShowUpdateConfirmModal] = useState(false);
  const [deleteModalMessage, setDeleteModalMessage] = useState(deleteMessage);
  const [showOverlay, setOverlay] = useState(false);

  const [requestError, setRequestError] = useState(false);
  const [requestSuccess, setRequestSuccess] = useState(false);
  const [requestLoading, setRequestLoading] = useState(false);
  const [requestMessage, setRequestMessage] = useState("");
  const [showRequestModal, setShowRequestModal] = useState(false);
  const [hideModal, setHideModal] = useState(false);
  const [requestAction, setRequestAction] = useState("");

  const inputsChangeHandler = (inputsObj, rowIndex, validInputs) => {
    setInputsState((prevState) =>
      prevState.map((obj, i) => {
        if (i === rowIndex) return { ...inputsObj };
        return obj;
      })
    );
    setInputsAreValid((prevState) =>
      prevState.map((obj, i) => {
        if (i === rowIndex) return { inputs: validInputs };
        return obj;
      })
    );
  };
  useEffect(() => {
    if (!validSubmit)
      setValidSubmit(
        inputsAreValid.every((isValid) => isValid.inputs === true)
      );
  }, [inputsAreValid, validSubmit]);
  const addNewRowHandler = () => {
    setInputsState((prevState) => {
      return (prevState = [...prevState, {}]);
    });
    setInputsAreValid((prevState) => {
      return (prevState = [...prevState, {}]);
    });
  };
  const removeRowHandler = (index) => {
    setInputsState((prevState) => {
      return [...prevState.filter((e, i) => i !== index)];
    });
    setInputsAreValid((prevState) => {
      return [...prevState.filter((e, i) => i !== index)];
    });
  };

  const submitItemHandler = async () => {
    setOverlay(false);
    const isValid = inputsAreValid.every((isValid) => isValid.inputs === true);
    setValidSubmit(isValid);
    if (isValid) {
      let typeId;
      let responseId = [];
      if (modalFor === "rooms") typeId = "roomId";
      if (modalFor === "courses") typeId = "courseId";
      if (modalFor === "teachers") typeId = "teacherId";
      setRequestAction("create");
      try {
        setShowUpdateConfirmModal(false);
        setHideModal(true);
        setShowRequestModal(true);
        setRequestLoading(true);
        for (let state of inputsState) {
          await axios.post(`/${modalFor}`, { ...state }, {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => {
            responseId.push({ [typeId]: response.data.id });
            if (modalFor === "subjects")
              onNewItem("subjectId", response.data.id);
            setResponseId(response.data.id);
          });
        }
      } catch (error) {
        setRequestLoading(false);
        setRequestError(true);
        setRequestMessage(content.errorMessages.requestAddError);
        return;
      }

      setRequestLoading(false);
      setRequestMessage(content.successMessages.create);
      setRequestSuccess(true);

      if (responseId.length > 0 && modalFor !== "subjects")
        onNewItem(modalFor, responseId);

      setInputsState([{}]);
      setInputsAreValid([{}]);
    }
  };

  const closeHandler = () => {
    if (modalFor === "subjects" && responseId) return onClose();
    if (responseId) return onClose();
    if (modalFor === "subjects") return onClose("subjectId");
    onClose(modalFor);
  };
  const confirmModalHandler = (event) => {
    if (event.target.name === "delete") {
      let roomBooked = scheduled.filter((row) => {
        if (row.rooms !== "") {
          let arr = row.rooms.filter(
            (room) => room.roomId === inputsState[0].id
          );
          return arr.length > 0 ? row : false;
        }
        return false;
      });
      let courseBooked = scheduled.filter((row) => {
        if (row.courses !== "") {
          let arr = row.courses.filter(
            (course) => course.courseId === inputsState[0].id
          );
          return arr.length > 0 ? row : false;
        }
        return false;
      });
      let teacherBooked = scheduled.filter((row) => {
        if (row.teachers !== "") {
          let arr = row.teachers.filter(
            (teacher) => teacher.teacherId === inputsState[0].id
          );
          return arr.length > 0 ? row : false;
        }
        return false;
      });
      let subjectBooked = scheduled.filter((row) => {
        if (row.subject !== "") {
          return row.subject.id === inputsState[0].id ? row : false;
        }
        return false;
      });

      if (modalFor === "rooms" && roomBooked.length > 0)
        setDeleteModalMessage(roomHasActiveTeachers);
      if (modalFor === "courses" && courseBooked.length > 0)
        setDeleteModalMessage(courseHasActiveTeachers);
      if (modalFor === "teachers" && teacherBooked.length > 0)
        setDeleteModalMessage(teacherHasActiveTeachers);
      if (modalFor === "subject" && subjectBooked.length > 0)
        setDeleteModalMessage(subjectHasActiveTeachers);

      setShowDeleteConfirmModal(true);
      setOverlay(true);
    }

    if (event.target.name !== "delete") {
      const isValid = inputsAreValid.every(
        (isValid) => isValid.inputs === true
      );
      if (!isValid) return setValidSubmit(isValid);
      if (event.target.name === "update") setShowUpdateConfirmModal(true);
      if (event.target.name === "create") setShowUpdateConfirmModal(true);
      setOverlay(true);
    }
  };

  const declineHandler = () => {
    setOverlay(false);
    setShowDeleteConfirmModal(false);
  };

  const declineUpdateHandler = () => {
    setOverlay(false);
    setShowUpdateConfirmModal(false);
  };

  const updateItemHandler = async () => {
    setOverlay(false);
    setRequestAction("update");
    try {
      setShowUpdateConfirmModal(false);
      setOverlay(false);
      setHideModal(true);
      setShowRequestModal(true);
      setRequestLoading(true);
      if (modalFor === "rooms") {
        await axios
          .patch(`/rooms/${editValues}`, inputsState[0], {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "courses") {
        await axios
          .patch(`/courses/${editValues}`, inputsState[0], {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "teachers") {
        await axios
          .patch(`/teachers/${editValues}`, inputsState[0], {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "subjects") {
        await axios
          .patch(`/subjects/${editValues}`, inputsState[0], {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
    } catch (error) {
      setRequestLoading(false);
      setRequestError(true);
      setRequestMessage(content.errorMessages.requestUpdateError);
      return;
    }

    setRequestLoading(false);
    setRequestMessage(content.successMessages.update);
    setRequestSuccess(true);
  };

  const deleteItemHandler = async () => {
    setRequestAction("delete");
    try {
      setShowDeleteConfirmModal(false);
      setOverlay(false);
      setHideModal(true);
      setShowRequestModal(true);
      setRequestLoading(true);
      if (modalFor === "rooms") {
        await axios
          .delete(`/rooms/${editValues}`, {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "courses") {
        await axios
          .delete(`/courses/${editValues}`, {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "teachers") {
        await axios
          .delete(`/teachers/${editValues}`, {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
      if (modalFor === "subjects") {
        await axios
          .delete(`/subjects/${editValues}`, {
            headers: { Authorization: `Bearer ${token.token}` },
          })
          .catch((error) =>  {
            if (error.response.status === 401) {
              setShowRequestModal(false);
              props.forceLogoutHandler();
              return error;
            }
            return error;
          })
          .then((response) => console.log(response));
      }
    } catch (error) {
      setRequestLoading(false);
      setRequestError(true);
      setRequestMessage(content.errorMessages.requestDeleteError);
      return;
    }
    setRequestLoading(false);
    setRequestMessage(content.successMessages.delete);
    setRequestSuccess(true);
    setDeleteModalMessage("KUSTUTA");
  };
  useEffect(() => {
    if (!validSubmit) {
      const timer = setTimeout(() => {
        setValidSubmit(true);
      }, 1000);
      return () => clearTimeout(timer);
    }
  }, [validSubmit]);

  useEffect(() => {
    if (requestSuccess) {
      const timer = setTimeout(() => {
        if (
          requestMessage === content.successMessages.delete ||
          requestMessage === content.successMessages.update
        ) {
          onDelete();
        }
        onClose();
        setHideModal(false);
        setShowRequestModal(false);
        setRequestSuccess(false);
        setRequestMessage("");
      }, 2000);
      return () => clearTimeout(timer);
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [requestSuccess]);

  const endRequestHandler = () => {
    onClose();
  };
  const failedRequestConfirmHandler = () => {
    setRequestError(false);
    setRequestMessage("");
    if (requestAction === "create") submitItemHandler();
    if (requestAction === "delete") deleteItemHandler();
    if (requestAction === "update") updateItemHandler();
  };

  return (
    <Modal onHidden={hideModal} onClose={closeHandler} overlay={showOverlay}>
      <div className="relative flex flex-col lg:pl-4">
        <div className="relative flex justify-end">
          <i
            onClick={closeHandler}
            className={`bi bi-x-lg absolute text-3xl lg:text-xl -top-2 -right-2`}
          ></i>
        </div>
        {modalFor === "subjects" &&
          inputsState.map((inputsRow, i) => {
            return (
              <div key={i}>
                <NewSubject
                  editValues={editValues}
                  editMode={editMode}
                  onAddNewRow={addNewRowHandler}
                  onRemoveRow={removeRowHandler}
                  modalFor={modalFor}
                  onChange={inputsChangeHandler}
                  index={i}
                  subjectsData={subjectsData}
                  values={inputsState[i]}
                  count={inputsState.length}
                />
              </div>
            );
          })}
        {modalFor === "teachers" &&
          inputsState.map((inputsRow, i) => {
            return (
              <div key={i}>
                <NewLecturer
                  editValues={editValues}
                  editMode={editMode}
                  onAddNewRow={addNewRowHandler}
                  onRemoveRow={removeRowHandler}
                  modalFor={modalFor}
                  onChange={inputsChangeHandler}
                  index={i}
                  teacherData={teacherData}
                  values={inputsState[i]}
                  count={inputsState.length}
                />
              </div>
            );
          })}
        {modalFor === "courses" &&
          inputsState.map((inputsRow, i) => {
            return (
              <div key={i}>
                <NewCourse
                  editValues={editValues}
                  editMode={editMode}
                  onAddNewRow={addNewRowHandler}
                  onRemoveRow={removeRowHandler}
                  modalFor={modalFor}
                  onChange={inputsChangeHandler}
                  index={i}
                  courseData={courseData}
                  values={inputsState[i]}
                  count={inputsState.length}
                />
              </div>
            );
          })}
        {modalFor === "rooms" &&
          inputsState.map((inputsRow, i) => {
            return (
              <div key={i}>
                <NewRoom
                  editValues={editValues}
                  editMode={editMode}
                  onAddNewRow={addNewRowHandler}
                  onRemoveRow={removeRowHandler}
                  modalFor={modalFor}
                  onChange={inputsChangeHandler}
                  index={i}
                  roomsData={roomsData}
                  values={inputsState[i]}
                  count={inputsState.length}
                />
              </div>
            );
          })}

        <div
          className={`flex ${
            editMode
              ? "justify-between space-x-20"
              : "justify-center lg:justify-between"
          } w-full pt-8`}
        >
          <div className="relative">
            {showDeleteConfirmModal && (
              <div className="absolute top-20 -left-16">
                <ConfirmModal
                  onDecline={declineHandler}
                  onConfirm={deleteItemHandler}
                  modalMessage={deleteModalMessage}
                  topArrow={true}
                />
              </div>
            )}
            {editMode && (
              <button
                onClick={confirmModalHandler}
                className="btn-actions"
                type="button"
                name="delete"
              >
                KUSTUTA
              </button>
            )}
          </div>
          <div className="relative">
            {!validSubmit && (
              <div className="absolute -top-16 -left-10 w-[150%]">
                <TooltipLarge message={mandatoryFields} />
              </div>
            )}

            <button
              onClick={confirmModalHandler}
              className="btn-actions"
              type="submit"
              name={editMode ? "update" : "create"}
            >
              SALVESTA
            </button>

            {validSubmit && showUpdateConfirmModal && (
              <div className="absolute top-20 -left-16">
                <ConfirmModal
                  onDecline={declineUpdateHandler}
                  onConfirm={editMode ? updateItemHandler : submitItemHandler}
                  modalMessage={saveMessage}
                  topArrow={true}
                />
              </div>
            )}
          </div>
        </div>
        {showRequestModal && (
          <RequestModal
            error={requestError}
            success={requestSuccess}
            loading={requestLoading}
            modalMessage={requestMessage}
            customStyle="top-1/2 lg:ml-32"
            onDecline={endRequestHandler}
            onConfirm={failedRequestConfirmHandler}
          />
        )}
      </div>
    </Modal>
  );
};

export default AddNewItem;
