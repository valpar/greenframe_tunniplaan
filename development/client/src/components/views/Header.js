import { ReactComponent as Logo } from "../../assets/logo/HK-est.svg";
import ConfirmModal from "../UI/ConfirmModal/ConfirmModal";

export const Header = (props) => {
  const {
    isTabletOrMobile,
    userPicture,
    onClick,  // see eemaldada
    loginMessage,
    loginInfo,
    showMobileFilters,
    admin,
    addScheduleHandler,
    scrollY,
    mobileFiltersHandler,
    mobileMenuHandler,
    showSchedule,
    hiddeMobileIcon,
    onUsersManagement,
    usersListOpen,
    filtersNotification,
    showMobilePicture,
    openModalAnimation,
    loginHandler,
  } = props;

  return (
    <header className="flex flex-col fixed pb-1 lg:pb-0 top-0 left-1/2 -translate-x-1/2 max-w-6xl w-full z-10 bg-white">
      <div className="flex items-center lg:items-end justify-between py-4 px-4 lg:py-0 lg:pt-4">
        {/* Logo */}
        <div className="w-40 h-full lg:w-80 lg:mb-4">
          <a
            href="https://www.tlu.ee/haapsalu"
            title="Avaleht"
            target="_blank"
            rel="noopener noreferrer"
          >
            <Logo />
          </a>
        </div>
        <div className="flex flex-row space-x-12">
          {/* Desktop menu */}
          <div className="hidden lg:flex flex-row justify-end items-end h-full font-serif mt-16">
            <div className="w-[0.1rem] h-7 bg-collegeGreen rotate-[15deg] mr-2 ml-3"></div>
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="https://start.hk.tlu.ee/sahtelbeta/sahtel/index.php"
              className="text-xl"
            >
              <i>SAHTEL</i>
            </a>
            <div className="w-[0.1rem] h-7 bg-collegeGreen rotate-[15deg] mr-2 ml-3"></div>
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
              className="text-xl"
            >
              <i>RIIUL</i>
            </a>
          </div>
          {/* Desktop login */}
          <div className="hidden relative lg:flex flex-col justify-end mb-1">
            <div className="mt-4 w-28 cursor-pointer" onClick={onClick}>
              <div className="mx-auto w-12 h-12">
                <img
                  src={userPicture}
                  alt="User"
                  className="w-full h-full rounded-full object-cover"
                ></img>
              </div>
              {/* <div className="text-lg mx-auto text-center">{userRoll}</div> */}
              <div className="text-lg mx-auto text-center">
                {loginMessage}
              </div>  
            </div>
          </div>
        </div>

        <div className="lg:hidden flex flex-row w-40 justify-end items-center space-x-7 pr-2">
          {/* Mobile schedule add */}
          {!usersListOpen && admin && isTabletOrMobile && !hiddeMobileIcon && (
            <i
              className={`bi bi-plus-lg text-3xl pt-[0.2rem] cursor-pointer ${
                scrollY < 766 && showSchedule ? "text-borderGray" : ""
              }`}
              onClick={addScheduleHandler}
            ></i>
          )}
          {/* Mobile filters */}
          <div className="relative pt-[0.1rem] pr-1">
            <i
              className={`${
                hiddeMobileIcon || usersListOpen ? "hidden" : ""
              } bi bi-sliders text-2xl cursor-pointer ${
                scrollY < 766 && showMobileFilters ? "text-borderGray" : ""
              }`}
              onClick={mobileFiltersHandler}
            ></i>
            {filtersNotification > 0 && <div className={`${
                hiddeMobileIcon || usersListOpen ? "hidden" : ""
              } absolute lg:hidden -top-1 -right-1 text-sm w-5 rounded-full bg-collegeGreen`}>
                {filtersNotification}
            </div>}
          </div>
          {/* Hamburger menu */}
          <button
            id="menu-btn"
            className={`${
              usersListOpen || (showMobilePicture && !openModalAnimation) ? "hidden" : ""
            } ${(showMobilePicture && openModalAnimation || !showMobilePicture && openModalAnimation) ? "open" : ""} hamburger focus:outline-none`}
            type="button"
            onClick={mobileMenuHandler}
          >
            <span className="hamburger-top"></span>
            <span className="hamburger-middle"></span>
            <span className="hamburger-bottom"></span>
          </button>
          {/* UserPicture */}
          {!usersListOpen && showMobilePicture && !openModalAnimation && <div className="mx-auto w-8 h-8" onClick={mobileMenuHandler}>
                <img
                  src={userPicture}
                  alt="User"
                  className="w-full h-full rounded-full object-cover"
                ></img>
          </div>}

          {usersListOpen && (
            <i
              onClick={onUsersManagement}
              className={`bi bi-x-lg text-[1.6em] mt-[0.1rem]`}
            ></i>
          )}

        </div>           
      </div>
      <div
        className={`w-full h-[0.2rem] bg-[#6c8298] border-solid border-1 border-[#d4d4d4]`}
      />
    </header>
  );
};
