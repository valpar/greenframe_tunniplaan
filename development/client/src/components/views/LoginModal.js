import React, { Fragment, useEffect } from "react";
import ReactDOM from "react-dom";
import Input from "../UI/Input/Input";

const Backdrop = (props) => {
  useEffect(() => {
    const themeColorMeta = document.querySelector('meta[name="theme-color"]');

    themeColorMeta.setAttribute("content", "#5C5C5C");

    return () => {
      themeColorMeta.setAttribute("content", "#fff");
    };
  }, []);

  return (
    <div
      data-testid="backdrop"
      className="fixed top-0 left-0 w-full h-screen z-20 bg-neutral-900 opacity-70"
      onClick={props.onClose}
    />
  );
};

const ConfirmModalOverlay = (props) => {
  
  const changeHandler = (input) => {
    
    if (input.name==="email") {
      props.onEmailChange(input.value);
    }
    if (input.name==="password") {
      props.onPasswordChange(input.value);
    }
  }
  

  let arrow =
    "absolute z-40 w-8 h-8 -rotate-45 -right-4 bg-white top-1/2 -translate-y-1/2";
  if (props.topArrow)
    arrow =
      "absolute z-40 w-8 h-8 -rotate-45 -top-4 bg-white left-1/2 -translate-x-1/2";
  if (props.bottomArrow)
    arrow =
      "absolute z-40 w-8 h-8 -rotate-45 -bottom-4 bg-white left-1/2 -translate-x-1/2";

  return (
    <div className="fixed top-1/2 -translate-y-1/2 left-1/2 -translate-x-1/2 lg:translate-x-0 lg:translate-y-0 lg:left-auto lg:top-auto lg:relative w-auto h-auto space-y-2 bg-white p-4 rounded z-30 shadow-lg">
      <div className={`hidden lg:block ${arrow}`}></div>
      <div className="pb-4 text-center">
        <p>{props.modalMessage}</p>
      </div>

      <label>
        E-mail: 
        <Input
          name="email" 
          value={props.email} 
          onChange={changeHandler} 
          readOnly={false}
          className={`p-2 border ${
            props.hasError || props.errorMessage
              ? "border-red-500"
              : "border-borderBlack "
          } rounded-none font-normal focus:outline-none focus:border-black`}
        />
      </label>
      <label>
        Parool: 
        <Input 
          name="password" 
          value={props.password} 
          onChange={changeHandler} 
          type="password" readOnly={false} 
          className={`p-2 border ${
            props.hasError || props.errorMessage
              ? "border-red-500"
              : "border-borderGray "
          } rounded-none font-normal focus:outline-none focus:border-black`}
        />
      </label>


      <div className="flex space-x-4">
        <button
          onClick={props.onDecline}
          type="button"
          className="border border-borderGray py-1 px-2 w-24 hover:bg-borderGray"
        >
          {props.modalButtons ? props.modalButtons[0] : `EI`}
        </button>
        <button
          onClick={props.onConfirm}
          type="button"
          className="border border-borderGray py-1 px-2 w-24 hover:bg-borderGray"
        >
          {props.modalButtons ? props.modalButtons[1] : `JAH`}
        </button>
      </div>
    </div>
  );
};



const portalElement = document.getElementById("overlays");

const LoginModal = (props) => {

  return (
    
    <Fragment>
      {ReactDOM.createPortal(<Backdrop />, portalElement)}
      <ConfirmModalOverlay {...props} /> 
    </Fragment>
    
  );
};

export default LoginModal;
