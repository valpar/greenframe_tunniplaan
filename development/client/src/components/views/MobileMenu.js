import SideModal from "../UI/Modal/SideModal";
import { useState } from "react";

const MobileMenu = (props) => {
  //const [showUserOptions, setShowUserOptions] = useState(false);
  const {
    onClose,
    loginHandler,
    loginMessage,
    loginInfo,
    userPicture,
    userAdminHandler,
    showMobileMenu,
    onUsersManagement,
    usersListOpen,
    admin,
  } = props;

  /*
  const loginHandler = () => {
    setShowUserOptions(true);
  };

  const userRollHandler = (e) => {
    setShowUserOptions(false);
    props.userRollHandler(e);
    onClose();
  };
*/
  const usersManagementHandler = () => {
    onUsersManagement();
    onClose();
  };

  return (
    <SideModal onHidden={showMobileMenu}>
      <div className={`flex flex-col`}>
        <div className="px-4 pt-8">
      {/* {!showUserOptions && ( */}
            <div className="flex flex-col space-y-8 w-full pb-10">
              {/* User photo */}
              <div className="mx-auto w-24 h-24">
                <img
                  src={userPicture}
                  alt="User"
                  className="w-full h-full rounded-full object-cover
            "
                ></img>
              </div>

              <p className="uppercase text-center">
                {loginInfo?.user?.firstName
                  ? loginInfo?.user?.firstName
                  : ""}
              </p>

              <button
                  onClick={loginHandler}
                  className="btn-m-menu"
                  type="button"
                  name="Login Google"
                >
                  {loginMessage}
              </button>
             

              <a
                className="btn-m-menu text-center"
                target="_blank"
                rel="noopener noreferrer"
                href="https://start.hk.tlu.ee/sahtelbeta/sahtel/index.php"
              >
                Sahtel
              </a>
              <a
                className="btn-m-menu text-center"
                target="_blank"
                rel="noopener noreferrer"
                href="https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
              >
                Riiul
              </a>
            </div>
   {/*    )}  */}
        </div>
      </div>
    </SideModal>
  );
};

export default MobileMenu;
