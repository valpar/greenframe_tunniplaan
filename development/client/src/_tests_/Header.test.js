import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Header } from '../components/views/Header';


describe('Header', () => {
    it("if matches snapshot", () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('displays admin controls when admin prop is true', () => {
        render(<Header admin={true} />);
        const adminControls = screen.getByTestId('admin-controls');
        expect(adminControls).toBeInTheDocument();
    });
});