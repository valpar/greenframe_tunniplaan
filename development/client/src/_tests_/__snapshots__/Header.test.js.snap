// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header if matches snapshot 1`] = `
<header
  className="flex flex-col fixed pb-1 lg:pb-0 top-0 left-1/2 -translate-x-1/2 max-w-6xl w-full z-10 bg-white"
>
  <div
    className="flex items-center lg:items-end justify-between py-4 px-4 lg:py-0 lg:pt-4"
  >
    <div
      className="w-40 h-full lg:w-80 lg:mb-4"
    >
      <a
        href="https://www.tlu.ee/haapsalu"
        rel="noopener noreferrer"
        target="_blank"
        title="Avaleht"
      >
        <svg>
          HK-est.svg
        </svg>
      </a>
    </div>
    <div
      className="flex flex-row space-x-12"
    >
      <div
        className="hidden lg:flex flex-row justify-end items-end h-full font-serif mt-16"
      >
        <div
          className="w-[0.1rem] h-7 bg-collegeGreen rotate-[15deg] mr-2 ml-3"
        />
        <a
          className="text-xl"
          href="https://start.hk.tlu.ee/sahtelbeta/sahtel/index.php"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i>
            SAHTEL
          </i>
        </a>
        <div
          className="w-[0.1rem] h-7 bg-collegeGreen rotate-[15deg] mr-2 ml-3"
        />
        <a
          className="text-xl"
          href="https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i>
            RIIUL
          </i>
        </a>
      </div>
      <div
        className="hidden relative lg:flex flex-col justify-end mb-1"
      >
        <div
          className="mt-4 w-28 cursor-pointer"
        >
          <div
            className="mx-auto w-12 h-12"
          >
            <img
              alt="User"
              className="w-full h-full rounded-full object-cover"
            />
          </div>
          <div
            className="text-lg mx-auto text-center"
          />
        </div>
      </div>
    </div>
    <div
      className="lg:hidden flex flex-row w-40 justify-end items-center space-x-7 pr-2"
    >
      <div
        className="relative pt-[0.1rem] pr-1"
      >
        <i
          className=" bi bi-sliders text-2xl cursor-pointer "
        />
      </div>
      <button
        className="  hamburger focus:outline-none"
        id="menu-btn"
        type="button"
      >
        <span
          className="hamburger-top"
        />
        <span
          className="hamburger-middle"
        />
        <span
          className="hamburger-bottom"
        />
      </button>
    </div>
  </div>
  <div
    className="w-full h-[0.2rem] bg-[#6c8298] border-solid border-1 border-[#d4d4d4]"
  />
</header>
`;
